extensions/fbc/src/org/sbml/jsbml/ext/fbc/converters/FbcV1ToCobraConverter.java

```diff
@@ -99,73 +217,28 @@ public SBMLDocument convert(SBMLDocument sbmlDocument) throws SBMLException {
-        kineticLaw.setMath(astNode);
-        LocalParameter fluxValueParameter = reaction.getKineticLaw().createLocalParameter(CobraConstants.FLUX_VALUE);
-        fluxValueParameter.setValue(0d);
-        fluxValueParameter.setUnits(CobraConstants.mmol_per_gDW_per_hr);
-        
-        // 3. Assign lower and upper flux bounds if they were set, use defaults otherwise
-        LocalParameter upperBoundParameter = reaction.getKineticLaw().createLocalParameter(CobraConstants.UPPER_BOUND);
-        if (reactionIdToUpperFluxBoundDoubleValueMap.containsKey(reaction.getId()))
-        {
-          upperBoundParameter.setValue(reactionIdToUpperFluxBoundDoubleValueMap.get(reaction.getId()).doubleValue());
-        } else if (this.defaultUpperBound != null)
-        {
-          upperBoundParameter.setValue(this.defaultUpperBound);
-        } else {
-          upperBoundParameter.setValue(Double.POSITIVE_INFINITY);
-        }
-        upperBoundParameter.setExplicitlyConstant(true);
-        upperBoundParameter.setUnits(CobraConstants.mmol_per_gDW_per_hr);
-        
-        LocalParameter lowerBoundParameter = reaction.getKineticLaw().createLocalParameter(CobraConstants.LOWER_BOUND);
-        if (reactionIdToLowerFluxBoundDoubleValueMap.containsKey(reaction.getId()))
-        {
-          lowerBoundParameter.setValue(reactionIdToLowerFluxBoundDoubleValueMap.get(reaction.getId()).doubleValue());
-        } else if (this.defaultLowerBound != null)
-        {
-          lowerBoundParameter.setValue(this.defaultLowerBound);  
-        } else {
-          lowerBoundParameter.setValue(Double.NEGATIVE_INFINITY);
-        }
-        lowerBoundParameter.setExplicitlyConstant(true);
-        lowerBoundParameter.setUnits(CobraConstants.mmol_per_gDW_per_hr);
-               
-        // 4. unset attribute constant for products and reactants
-        for (SpeciesReference speciesReference: reaction.getListOfProducts()) {
-          speciesReference.unsetConstant();
-        }
-        for (SpeciesReference speciesReference: reaction.getListOfReactants()) {
-          speciesReference.unsetConstant();
-        }
-        
+      // update lower and upper bound for existing fluxBounds
+      for (FluxBound fluxBound : fbcModelPlugin.getListOfFluxBounds()) {
+        this.updateKineticLawFromBound(model.getReaction(fluxBound.getReaction()), fluxBound);
       }
-      // 5. unset List of flux bounds
-      fbcModelPlugin.unsetListOfFluxBounds();
+      
+      // set objectiveCoefficient values for active objective
+      this.setObjectiveCoefficient(fbcModelPlugin, model);
     }
    
-    // TODO: it is unclear whether this should be done
-    sbmlDocument.removeDeclaredNamespaceByPrefix(FBCConstants.shortLabel);
-    // TODO: the SBMLDocumentAttriute fbc:required="false" still remains and it is unclear how to remove it
+    // remove FBC Package
+    packageDisabler.removePackage(FBCConstants.shortLabel);
```

Comment:
Now the PackageDisabler is being used to remove the fbc package from the converted model.